x/evm/keeper/grpc_query.go

```diff
@@ -341,8 +352,28 @@ func (k Keeper) EstimateGas(c context.Context, req *types.EthCallRequest) (*type
 			msg.IsFake(),
 		)
 
+		tmpCtx := ctx
+		if fromType == FromTypeRPC {
+			ctx = ctx.WithGasMeter(sdk.NewGasMeter(msg.Gas())).WithKVGasConfig(storetypes.GasConfig{}).
```

Comment:
Please use `NewInfiniteGasMeterWithLimit` instead, same as we're doing [in traceTx](https://github.com/evmos/evmos/blob/d519cecde0c5320437f4ec09d71c6763d8473d8f/x/evm/keeper/grpc_query.go#L443)
```suggestion
			ctx = ctx.WithGasMeter(evmostypes.NewInfiniteGasMeterWithLimit(msg.Gas())).WithKVGasConfig(storetypes.GasConfig{}).
```